tracer/src/Datadog.Trace/LifetimeManager.cs

```diff
@@ -94,6 +96,13 @@ private void CurrentDomain_DomainUnload(object sender, EventArgs e)
             AppDomain.CurrentDomain.DomainUnload -= CurrentDomain_DomainUnload;
         }
 
+        private void CiVisibilityExit(object sender, EventArgs e)
+        {
+            Log.Information("CI Visibility is exiting.");
+            RunShutdownTasks();
+            CIVisibility.CIVisibilityExit -= CiVisibilityExit;
+        }
+
```

Comment:
I dunno, I would probably have preferred to just make `RunShutdownTasks()` public, and invoke it directly from `CIVisibility.Close()`. This approach seems to creates a slightly odd circular dependency IMO: I don't think the `LifetimeManager` should depend on `CIVisibility`...

Reply:
done in [1b3135a](https://github.com/DataDog/dd-trace-dotnet/pull/4671/commits/1b3135ab40bb99786fc707f1bc825abe1f8126b0)